internal/pusher/pusher.go

```diff
@@ -128,8 +142,14 @@ func (p *Publisher) publish(ctx context.Context, payload Payload) {
 				logger.Error().Err(err).Msg("publish events")
 				p.errorCounter.WithLabelValues("logs", tenantStr, strconv.Itoa(httpStatusCode)).Inc()
 				if hasStatusCode && httpStatusCode == http.StatusUnauthorized {
-					// Retry to get a new client, credentials might be stale.
+					// Retry with backoff to get a new client, credentials might be stale.
 					newClient = true
+
+					retries--
+					if retries > 0 {
+						time.Sleep(backoff)
```

Comment:
after looking more into the code, maybe we need to think about this, and instead of putting in a retry with backoff, we can do a single retry and give up. 

we are already giving up when we get an error while calling `getClient`, so if we get a invalid client with `newClient: true`, we will probably not get a valid again on second try. 

`publish` method is called in a new goroutine, and caller spins up a new goroutines to publish data from each check run. So If we sleep for a lot of time, a bad check can lead to goroutines piling up.